tokio-util/src/task/join_map.rs

```diff
@@ -316,6 +316,50 @@ where
         self.insert(key, task);
     }
 
+    /// Spawn the blocking code on the blocking threadpool and store it in this `JoinMap` with the provided
+    /// key.
+    ///
+    /// If a task previously existed in the `JoinMap` for this key, that task
+    /// will be cancelled and replaced with the new one. The previous task will
+    /// be removed from the `JoinMap`; a subsequent call to [`join_next`] will
+    /// *not* return a cancelled [`JoinError`] for that task.
```

Comment:
we should maybe mention that once a blocking task begins executing, it will still run to completion even if cancelled, here?

Reply:
Added a comment that should explain the cancellation behaviour.